pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -0,0 +1,289 @@
+		Name:      secretName,
+		Namespace: secretNamespace,
+	}}
+	obj.Status.VPCPath = vpcPath
+	// TODO: Add proxy
+	return s.Client.Status().Update(ctx, obj)
+}
+
+func (s *NSXServiceAccountService) DeleteNSXServiceAccount(ctx context.Context, namespacedName types.NamespacedName) error {
+	clusterName := s.getClusterName(namespacedName.Namespace, namespacedName.Name)
+	normalizedClusterName := util.NormalizeId(clusterName)
+	// delete Secret
+	secretName := namespacedName.Name + SecretSuffix
+	secretNamespace := namespacedName.Namespace
+	if err := s.Client.Delete(ctx, &v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: secretNamespace}}); err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Failed to delete", "secret", secretName, "namespace", secretNamespace)
+		return err
+	}
+
+	// delete ClusterControlPlane
+	cascade := true
+	if err := s.NSXClient.ClusterControlPlanesClient.Delete(siteId, enforcementpointId, normalizedClusterName, &cascade); err != nil {
+		log.Error(err, "Failed to delete", "ClusterControlPlane", normalizedClusterName)
+		return err
+	}
+	s.ClusterControlPlaneStore.Delete(model.ClusterControlPlane{Id: &normalizedClusterName})
+
+	// delete PI
+	if piobj := s.PrincipalIdentityStore.GetByKey(normalizedClusterName); piobj != nil {
+		pi := piobj.(mpmodel.PrincipalIdentity)
+		if err := s.NSXClient.PrincipalIdentitiesClient.Delete(*pi.Id); err != nil {
+			log.Error(err, "Failed to delete", "PrincipalIdentity", *pi.Name)
+			return err
+		}
+		if pi.CertificateId != nil && *pi.CertificateId != "" {
+			if err := s.NSXClient.CertificatesClient.Delete(*pi.CertificateId); err != nil {
+				log.Error(err, "Failed to delete", "PrincipalIdentity", *pi.Name, "Certificate", *pi.CertificateId)
+				return err
+			}
+		}
+		s.PrincipalIdentityStore.Delete(pi)
+	}
+	return nil
+}
+
+// ListNSXServiceAccountRealization returns all existing realized or failed NSXServiceAccount on NSXT
+func (s *NSXServiceAccountService) ListNSXServiceAccountRealization() sets.String {
+	// List PI
+	uidSet := s.PrincipalIdentityStore.ListIndexFuncValues(common.TagScopeNSXServiceAccountCRUID)
+
+	// List ClusterControlPlane
+	uidSet = uidSet.Union(s.ClusterControlPlaneStore.ListIndexFuncValues(common.TagScopeNSXServiceAccountCRUID))
+	return uidSet
+}
+
+func (s *NSXServiceAccountService) GetNSXServiceAccountNameByUID(uid string) (namespacedName types.NamespacedName) {
+	objs, err := s.PrincipalIdentityStore.ByIndex(common.TagScopeNSXServiceAccountCRUID, uid)
+	if err != nil {
+		log.Error(err, "Failed to search PrincipalIdentityStore by UID")
+		return
+	}
+	for _, obj := range objs {
+		pi := obj.(mpmodel.PrincipalIdentity)
+		for _, tag := range pi.Tags {
+			switch *tag.Scope {
+			case common.TagScopeNamespace:
+				namespacedName.Namespace = *tag.Tag
+			case common.TagScopeNSXServiceAccountCRName:
+				namespacedName.Name = *tag.Tag
+			}
+			if namespacedName.Name != "" && namespacedName.Namespace != "" {
+				return
```

Comment:
we do not need return namespaceName?

Reply:
In line 243, we described the return value name is `namespacedName`, thus we can return it with a simple `return` command without parameters.

Reply:
ok ,thanks.